taichi/transforms/offload.cpp

```diff
@@ -259,7 +259,6 @@ class IdentifyValuesUsedInOtherOffloads : public BasicStmtVisitor {
   }
 
   std::size_t allocate_global(DataType type) {
-    TI_ASSERT(type.width == 1);
```

Comment:
Could you please explain why you removed this line?

Reply:
Good question! This is simply because the `allocate_global` function should work for vectorized types in the future, since  `DataType` maybe a `VectorType`.

Since this has led to confusion, I added back an `TI_ASSERT(type->vector_width() == 1);` statement there following the new API.

Reply:
Thanks for your detailed explanation! Then I have no question now and I will review the changes again.